### PR TITLE
fix: do not run CHECKPOINT as streaming_replica user

### LINKED ISSUE #1407: Creating a new `Backup` with `immediate` `true` fails because the `streaming_replica` user doesn't have rights to do `CHECKPOINT`
If I create a new Backup with `immediate` `true` it fails because the `streaming_replica` user doesn't have rights to do `CHECKPOINT`, see the following logs:

```code
{"level":"info","ts":"2023-02-07T11:11:03Z","msg":"WAL archiving is not working","logging_pod":"cluster-name-1","err":"error while requiring a checkpoint: ERROR: must be superuser or have privileges of pg_checkpoint to do CHECKPOINT (SQLSTATE 42501)"}
```

### PR DESCRIPTION
**Pull request by mnencia**
Closes #1407

### CODE REVIEW

**Review by armru — APPROVED** [review]

File: pkg/management/postgres/wal.go [hunk 1]
```diff
@@ -28,63 +28,41 @@ import (
...
+func (w *walArchiveAnalyzer) execute() error {
+	return retry.OnError(w.backoff, resources.RetryAlways, func() error {
+		db, err := w.dbFactory()
+		if err != nil {
+			return err
 		}
-		return db, nil
+		return w.mustHaveFirstWalArchived(db)
 	})
 }
```
Comment (phisco) [inline pkg/management/postgres/wal.go:63]: this connection is never closed, is that intentional? @armru

Comment (armru) [reply pkg/management/postgres/wal.go:63]: yes this was intentional in the old implementation, but I've streamlined the logic further

File: pkg/management/postgres/wal.go [hunk 2]
```diff
@@ -97,20 +98,74 @@ func (w *walArchiveBootstrapper) tryBootstrapWal() error {
...
-	if w.firstWalArchiveTriggered {
-		log.Info("Waiting for the first WAL file to be archived")
-		return errors.New("waiting for first wal-archive")
+	return errNoWalArchivePresent
+}
+
+type walArchiveBootstrapper struct {
+	walArchiveAnalyzer
+	createdFirstWal bool
+}
```
Comment (jsilvela) [inline pkg/management/postgres/wal.go:116]: I think this could use some documentation

File: pkg/management/postgres/wal.go [hunk 3]
```diff
@@ -97,20 +98,74 @@ func (w *walArchiveBootstrapper) tryBootstrapWal() error {
...
+				log.Debug("Error while closing connection", "err", closeErr.Error())
+			}
+		}()
+
+		err = w.mustHaveFirstWalArchived(db)
+		if !errors.Is(err, errNoWalArchivePresent) {
+			return err
+		}
+
+		if w.createdFirstWal {
```
Comment (jsilvela) [inline pkg/management/postgres/wal.go:157]: I think this is very wrong...
And, I don't see where it gets set (other than in test)

Comment (jsilvela) [reply pkg/management/postgres/wal.go:157]: I think that in the shipWalFile method, we should set `w.createdFirstWal` to true, no?

**Review by jsilvela — APPROVED** [review]
Looks good, but have found the code difficult to parse initially.
We should open a new ticket to simplify.
And, just a personal thing, but `dbFactory` unsettles me.

**Pull request merged**